main.cpp

```diff
@@ -13,75 +13,505 @@ What you create in this project will be used as the basis of Project 5 in the co
+    int numberOfCarsServiced = 10;               
     
-Sub Object 3: Environment Controls
-    Name 3 things you'll find on the:    Environment Controls
-        1)
-        2)
-        3)
-    Name 3 things you can do with the:   Environment Controls
-        1)
-        2)
-        3)
-
-Sub Object 4: Infotainment System
-    Name 3 things you'll find on the:    Infotainment System
-        1)
-        2)
-        3)
-    Name 3 things you can do with the:   Infotainment System
-        1)
-        2)
-        3)
-
-Sub Object 5: Seat 
-    Name 3 things you'll find on the:    Seat
-        1)
-        2)
-        3)
-    Name 2 things you can do with the:   Seat
-        1)
-        2)
-*/
+    struct Car //5)                                 Note that the nested type 'Car' is related to the 'CarWash' 
+    {
+        //2) member variables with relevant data types.  the names are appropriate for the U.D.T.'s purpose.
+        bool isAPickupTruck = false;
+        float gasMileage = 26.2f;        
+        int year = 1985;
+        std::string manufacturer = "Toyota";
+        std::string model = "Corolla";
+
+        //3) a member function whose parameter has a default value.
+        //the parameter name is related to the work the function will perform.
+        void fillTank(double fuelAmountInGallons = 2.0);  
+        void breakDown(std::string failureType, bool requiresTow = false);
+        int getMilesTraveledAnnually(bool includeUberLyftTrips);
+    };
+
+    //wash and wax car
+    void washAndWaxCar(Car car); //4) a member function whose parameter is a UDT.
+    //charge customer
+    float chargeCustomer(float discountPercentage);
+    //detail the car interior
+    void detailInterior(Car car);
+    
+    //5) a member variable whose type is a UDT.
+    Car carBeingServiced;  
+};
+
+
+/*
+Thing 1) electric Guitar
+5 properties:
+    1) number of strings (int)
+    2) number of pickups (int)
+    3) color (std::string)
+    4) age (int)
+    5) price (float)
+3 things it can do:
+    1) generate a Note
+    2) break a String
+    3) tune
+ */
+
+struct ElectricGuitar
+{
+    // number of strings (int)
+    int numberOfStrings = 6;
+    // number of pickups (int)
+    int numberOfPickups = 2;
+    // color (std::string)
+    std::string color = "blue";
+    // age (int)
+    int age = 12;
+    // price (float)
+    float price = 1299.99f;
+
+    struct GuitarString
+    {
+        std::string manufacturer = "Ernie Ball";
+        int number = 0;
+        bool isWound = true;
+        float width = 0.52f;
+        std::string material = "steel";
+
+        void breakString();
+        void tune (float pitch);
+        void pluck(int fret); 
+    };
+
+    // generate a Note
+    void generateNote(GuitarString string, int fretPosition);
+    // break a String
+    void breakString(GuitarString stringNumber);
+    // tune
+    void tune(GuitarString stringNumber, float referencePitch); 
+
+    GuitarString string; 
```

Comment:
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#).
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in.
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends.
the C++ concept of References and Pointers has not been covered at this point in the course.  If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.